outlines/text/generate/sequence.py

```diff
@@ -40,6 +40,38 @@ def is_finished(self, token_ids: torch.LongTensor) -> torch.BoolTensor:
     def postprocess_completions(self, completions: List[str]) -> List[str]:
         return completions
 
+    def _next_token_logits(
+        self,
+        num_prompt_tokens: int,
+        token_ids: torch.LongTensor,
+        attention_mask: torch.LongTensor,
+    ) -> torch.FloatTensor:
+        logits = self.model(token_ids, attention_mask)
+        logits = self.create_proposal(token_ids[:, num_prompt_tokens:], logits)
+        return logits
+
+    def next_token_logits(self, prompt: Union[str, List[str]]) -> torch.FloatTensor:
+        """Compute the next-token logits for a given prompt.
+
+        Parameters
+        ----------
+        prompt
+            The input prompt.
+
+        Returns
+        -------
+        An array of shape `(batch_size, vocab_size)` containing the logits
+        (unnormalised log probabilities) for the next token generation.
+
+        """
+        token_ids, attention_mask = self.model.tokenizer.encode(prompt)
+        num_prompt_tokens = token_ids.shape[-1]
```

Comment:
I'm wondering if this is working as intended in the vectorised case.

Surely in the vectorised case, different prompts will have different lengths

Reply:
This was my bad, indeed the attention_mask handles it.

I was getting confused seeing different probs on different runs but it was because I was using GPT2 which has Dropout